ouroboros-consensus-protocol/src/ouroboros-consensus-protocol/Ouroboros/Consensus/Protocol/Praos/Common.hs

```diff
@@ -54,36 +55,178 @@ data PraosChainSelectView c = PraosChainSelectView
   }
   deriving (Show, Eq, Generic, NoThunks)
 
+-- | When to compare the VRF tiebreakers.
+data VRFTiebreakerFlavor =
+    -- | Always compare the VRF tiebreakers. This is the behavior of all eras
+    -- before Conway. Once mainnet has transitioned to Conway, we can remove
+    -- this option.
+    UnrestrictedVRFTiebreaker
+  | -- | Only compare the VRF tiebreakers when the slot numbers differ by at
+    -- most the given number of slots.
+    --
+    -- The main motivation is as follows:
+    --
+    -- When two blocks A and B with the same block number differ in their slot
+    -- number by more than Δ (the maximum message delay from Praos), say
+    -- @slot(A) + Δ < slot(B)@, the issuer of B should have been able to mint a
+    -- block with a block number higher than A (eg by minting on top of A).
+    -- Therefore, we do not want to allow B to win against A by having a better
```

Comment:
It occurs to me that today's excessively granular tiebreakers mean that short forks resolve themselves ASAP on mainnet. As of this change, the network instead won't resolve until someone creates a sole longest chain again (ie the actual Praos paper). I wonder if the community needs to be "warned" about short forks becoming "relatively long lasting" after this change?

Reply:
> There's also a scenario in which case A was the poorly managed pool. Perhaps B didn't extend A because A was very late sending their block. The tiebreaker still works in that case, since B would (likely) have arrived to the network before A.

Good point, will expand this to make it clear that both A and B could be the culprit here.

> It occurs to me that today's excessively granular tiebreakers mean that short forks resolve themselves ASAP on mainnet. As of this change, the network instead won't resolve until someone creates a sole longest chain again (ie the actual Praos paper). I wonder if the community needs to be "warned" about short forks becoming "relatively long lasting" after this change?

Yeah, that is an advantage of the current "deterministic"[^det]/"consistent"[^con] tiebreaker. I think it makes sense to mention that in https://github.com/IntersectMBO/ouroboros-network/issues/2913 when the HF approaches :+1:

[^det]: That's how it is called eg in [CAD-2545](https://input-output.atlassian.net/browse/CAD-2545).
[^con]: That's how it is called in [this paper](https://eprint.iacr.org/2020/041.pdf) that the researchers referred us to when we talked about this; the high-level summary back then was that you get *slightly* better settlement bounds with a consistent tiebreaker. The paper only seems to prove a similar asymptotic bound under weaker conditions (Theorem 2 vs Theorem 1); but this is fully consistent that.